docs/snapshotting/random-for-clones.md

```diff
@@ -0,0 +1,233 @@
+4.19 (or higher), we can rely on the `CONFIG_RANDOM_TRUST_CPU` kernel
+option (or the random.trust_cpu=on cmdline parameter) to have the
+entropy pool automatically refilled using the `CPU HWRNG`, so step 3
+would no longer be necessary. Another way around step 3 is to attach a
+`virtio-rng` device, but that’s not currently supported by Firecracker.
+
+## Annex 1: Source code that clears and reinitializes the entropy pool
+
+```cpp
+#include <errno.h>
+#include <fcntl.h>
+#include <linux/random.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+
+void exit_usage() {
+    printf("Usage: ./rerand [<hexadecimal_string>]\n"
+           "The length of the string must be a multiple of 8.\n");
+    exit(EXIT_FAILURE);
+}
+
+void exit_perror(const char *msg) {
+    perror(msg);
+    exit(EXIT_FAILURE);
+}
+
+int main(int argc, char ** argv) {
+    if (argc > 2) {
+        exit_usage();
+    }
+
+    size_t len = 0;
+    struct rand_pool_info *info = NULL;
+
+    if (argc == 2) {
+        len = strlen(argv[1]);
+        // We want len to be a multiple of 8 such that we have an easier time
+        // parsing argv[1] into an array of u32s.
+        if (len % 8) {
+            exit_usage();
+        }
+
+        info = malloc(sizeof(struct rand_pool_info) + len / 8);
+        if (info == NULL) {
+            exit_perror("Could not alloc rand_pool_info struct");
+        }
+        // This is measured in bits IIRC.
+        info->entropy_count = len * 4;
+        info->buf_size = len / 8;
+    }
+
+    int fd = open("/dev/urandom", O_RDWR);
+    if (fd < 0) {
+        exit_perror("Unable to open /dev/urandom");
+    }
+
+    if (ioctl(fd, RNDCLEARPOOL) < 0) {
+        exit_perror("Error issuing RNDCLEARPOOL operation");
+    }
+
+    if (argc == 1) {
+        exit(EXIT_SUCCESS);
+    }
+
+    // Add the entropy bytes supplied by the user.
+    char num_buf[9] = {};
+    size_t pos = 0;
+
+    while (pos < len) {
+        memcpy(num_buf, &argv[1] + pos, 8);
+        info->buf[pos / 8] = strtoul(num_buf, NULL, 16);
+        pos += 8;
+    }
+
+    if (ioctl(fd, RNDADDENTROPY, info) < 0) {
+        exit_perror("Error issuing RNDADDENTROPY operation");
+    }
+}
+```
+
+## References
+
+[1] http://man7.org/linux/man-pages/man7/random.7.html (please note the
+    online man pages are specific to a particular version, mentioned at
+    the end; also, for some reason, this man page is not available by
+    default on all distros)
+
+[2] https://www.2uo.de/myths-about-urandom
+
+[3] https://www.bsi.bund.de/SharedDocs/Downloads/EN/BSI/Publications/Studies/LinuxRNG/LinuxRNG_EN.pdf
+
+[4] http://man7.org/linux/man-pages/man4/random.4.html
+
+[1]: http://man7.org/linux/man-pages/man7/random.7.html
+[2]: https://www.2uo.de/myths-about-urandom
+[3]: https://www.bsi.bund.de/SharedDocs/Downloads/EN/BSI/Publications/Studies/LinuxRNG/LinuxRNG_EN.pdf
+[4]: http://man7.org/linux/man-pages/man4/random.4.html
```

Comment:
```suggestion
[1] http://man7.org/linux/man-pages/man7/random.7.html (please note the
    online man pages are specific to a particular version, mentioned at
    the end; also, for some reason, this man page is not available by
    default on all distros)
[2] https://www.2uo.de/myths-about-urandom
[3] https://www.bsi.bund.de/SharedDocs/Downloads/EN/BSI/Publications/Studies/LinuxRNG/LinuxRNG_EN.pdf
[4] http://man7.org/linux/man-pages/man4/random.4.html
```

Reply:
When opening it with a markdown interpreter the second batch of links the ones with [$NR]: are used to generate the links and are a mode to not specify links in-place. Otherwise the lines would extend pass the 80 char limit. Please see how this looks in my branch.